VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -182,30 +184,49 @@ def _dict_to_name_value_array(self, values):
                     })
         return array
 
-    def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, kek_algorithm, extra_device_items, disk_util, crypt_mount_config_util):
+    def get_settings_data(self, protector_name, kv_url, kv_id, kek_url, kek_kv_id, kek_algorithm, extra_device_items, disk_util, crypt_mount_config_util, key_store_type, keystoretype_flag_exists):
         """ returns encryption settings object in format required by wire server """
 
-        # validate key vault parameters prior to creating the encryption settings object
         cutil = CheckUtil(self.logger)
-        cutil.check_kv_id(kv_id, "A KeyVault ID is required, but is missing or invalid")
-        cutil.check_kv_url(kv_url, "A KeyVault URL is required, but is missing or invalid")
-        cutil.check_kv_name(kv_id, kv_url, "A KeyVault ID and KeyVault URL were provided, but their key vault names did not match")
-        if kek_url:
-            cutil.check_kv_id(kek_kv_id, "A KEK URL was specified, but its KEK KeyVault ID was missing or invalid")
-            cutil.check_kek_url(kek_url, "A KEK URL was specified, but it was invalid")
-            cutil.check_kek_name(kek_kv_id, kek_url, "A KEK ID and KEK URL were provided, but their key vault names did not match")
-            if kek_algorithm not in CommonVariables.encryption_algorithms:
-                if kek_algorithm:
-                    raise Exception("The KEK encryption algorithm requested was not recognized")
-                else:
-                    kek_algorithm = CommonVariables.default_encryption_algorithm
-                    self.logger.log("No KEK algorithm specified, defaulting to {0}".format(kek_algorithm))
+
+        # Perform algorithm check regaurdless of ManagedHSM or Keyvault
+        if kek_algorithm not in CommonVariables.encryption_algorithms:
+                    if kek_algorithm:
+                        raise Exception("The KEK encryption algorithm requested was not recognized")
+                    else:
+                        kek_algorithm = CommonVariables.default_encryption_algorithm
+                        self.logger.log("No KEK algorithm specified, defaulting to {0}".format(kek_algorithm))
+
+        # ManagedHSM Checks
+        if key_store_type and key_store_type.lower() == CommonVariables.KeyStoreTypeManagedHSM.lower():
```

Comment:
Instead of 
`if key_store_type and key_store_type.lower() == CommonVariables.KeyStoreTypeManagedHSM.lower():`

any reason we can't we have 
`if key_store_type.lower() == CommonVariables.KeyStoreTypeManagedHSM.lower():` ?

Reply:
I can't remember how python would treat calling lower() on a null value. I think I put it there so that it does not try to call lower() if the key_store_type is null, causing an exception. I can test, rusty Python knowledge :)

Reply:
Cool. I am okay if that's the case

Reply:
Confirmed this Via a separate error, from some other unrelated testing. When calling .lower() on a null value: 

2023/03/10 01:43:04 [Microsoft.Azure.Security.Edp.AzureDiskEncryptionForLinux-1.0]: [StatusReport (0)] msg: Failed to enable the extension with error: 'NoneType' object has no attribute 'lower', stack trace: Traceback (most recent call last):
2023/03/10 01:43:04   File "/var/lib/waagent/Microsoft.Azure.Security.Edp.AzureDiskEncryptionForLinux-1.2.0.104/main/handle.py", line 2321, in daemon
2023/03/10 01:43:04     daemon_encrypt()
2023/03/10 01:43:04   File "/var/lib/waagent/Microsoft.Azure.Security.Edp.AzureDiskEncryptionForLinux-1.2.0.104/main/handle.py", line 1969, in daemon_encrypt
2023/03/10 01:43:04     encryption_cmd = encryption_marker.**get_current_command().lower()**
2023/03/10 01:43:04 **AttributeError: 'NoneType' object has no attribute 'lower'**